### PR TITLE
[DON"T REVIEW]TW-459: reduce frames in chat list

### PR DESCRIPTION
**Pull request by sherlockvn**
### Before refactoring:

https://github.com/linagora/twake-on-matrix/assets/43041967/ca29e492-5cef-4f90-a912-82714f135067

### After refactoring:

https://github.com/linagora/twake-on-matrix/assets/43041967/c28981d0-0dc5-4afa-9a5d-490fe5bbad9f

### Conclusion:
The average and max ms/frame before refactoring is lower than after refactoring, so the performance before is better

### CODE REVIEW

File: lib/pages/chat_list/chat_list.dart [hunk 1]
```diff
@@ -79,6 +78,8 @@ class ChatListController extends State<ChatList>
 
   String? activeSpaceId;
 
+  final chatListIndexes = <String, int>{};
```
Comment (hoangdat) [inline lib/pages/chat_list/chat_list.dart:81]: what is it

Comment (sherlockvn) [reply lib/pages/chat_list/chat_list.dart:81]: It's a method for optimize the chat list when chat list is reordered

**Comment by sherlockvn** [comment]
> Your PR is very very hard for me.

describe your problem

**Comment by nqhhdev** [comment]
>  ADR + video Demo

File: lib/pages/chat_list/chat_list_body.dart [hunk 2]
```diff
@@ -31,224 +28,200 @@ class ChatListViewBody extends StatelessWidget {
     final userSearchResult = controller.userSearchResult;
     final client = Matrix.of(context).client;
 
-    return PageTransitionSwitcher(
```
Comment (nqhhdev) [inline lib/pages/chat_list/chat_list_body.dart:34]: Why did you delete it? It's better about performance

Comment (sherlockvn) [reply lib/pages/chat_list/chat_list_body.dart:34]: Why it's better for performance ?. I think currently we don't use the animation, so just remove it for now
